File: api/internal/activity/entity.go
package activity

import (
	"time"

	"github.com/google/uuid"
)

type Activity struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Emoji       string    `json:"emoji"`
	Badge       string    `json:"badge"`
	PhotoURL    *string   `json:"photo_url"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: web/modules/activity/entity.ts
export type Activity = {
  id: string;
  title: string;
  description: string;
  emoji: string;
  badge: string;
  photo_url: string | null;
  youtube_url: string | null;
  created_at: string;
  updated_at: string;
};

export type NewActivity = Omit<Activity, "id" | "created_at" | "updated_at">;
